### PR TITLE
Feature merkle trie proof

### PR DESCRIPTION
**Pull request by sircoon4**
- Added merkle trie inclusion proof for libplanet as precompiled contract.
- Included tests for 4 success cases and 4 failure(return false) cases.

### CODE REVIEW

File: core/vm/contracts.go [hunk 1]
```diff
@@ -82,15 +83,16 @@ var PrecompiledContractsIstanbul = map[common.Address]PrecompiledContract{
...
+	common.BytesToAddress([]byte{1}):          &ecrecover{},
+	common.BytesToAddress([]byte{2}):          &sha256hash{},
+	common.BytesToAddress([]byte{3}):          &ripemd160hash{},
+	common.BytesToAddress([]byte{4}):          &dataCopy{},
+	common.BytesToAddress([]byte{5}):          &bigModExp{eip2565: true},
+	common.BytesToAddress([]byte{6}):          &bn256AddIstanbul{},
+	common.BytesToAddress([]byte{7}):          &bn256ScalarMulIstanbul{},
+	common.BytesToAddress([]byte{8}):          &bn256PairingIstanbul{},
+	common.BytesToAddress([]byte{9}):          &blake2F{},
+	common.BytesToAddress([]byte{0x02, 0x00}): &libplanetVerifyProof{},
```
Comment (eseiker) [inline core/vm/contracts.go]: while our devnet/testnet are on op fjord hardfork now, i think this precompile address should be enabled on fjord.

File: core/vm/testdata/precompiles/libplanetVerifyProof.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "Expected": "0000000000000000000000000000000000000000000000000000000000000000",
+    "Name": "sampleFalseProof",
+    "Gas": 3000
+  }
+]
```
Comment (eseiker) [inline core/vm/testdata/precompiles/libplanetVerifyProof.json:50]:
```suggestion
]

```

File: go.mod
Comment (eseiker) [inline go.mod:1]: to avoid unexpected behavior and for ease of maintainability, i hope to stick with same go and deps version with upstream

**Review by eseiker — APPROVED** [review]
LGTM with some comments

+ not sure core/vm/libplanet is fine path
+ seems there's a convention with RequiredGas to be placed in protocol_params.go
https://github.com/planetarium/op-geth/blob/4a64d623a0f5e30be372f692472afff23d434d45/core/vm/contracts.go#L1197
https://github.com/planetarium/op-geth/blob/4a64d623a0f5e30be372f692472afff23d434d45/params/protocol_params.go#L169

**Comment by sircoon4** [comment]
@eseiker 
I moved the libplanet package location and followed a convention for RequiredGas.

**Review by ost006 — APPROVED** [review]
LGTM

**Pull request merged**